### PR TITLE
[gitlab] update builderimages for go 1.19.4 builds

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?

Bumps buildimages to new iteration shipping go `1.19.4`.

### Motivation

Time to hop on `1.19.4` as EOL for 1.18 approaches fast.

### Additional Notes

Please note, there's a large number of `gofmt` changes that had to be made as well to adhere to updated to formatting convention with `1.19.4`. 

### Possible Drawbacks / Trade-offs

Should be a NOP, but important QA scrutiny to avoid introducing potential regressions (RSS, cpu/threading, GC behavior, etc). 

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

**Review by guyarb — APPROVED** [review]
looks for USM

File: releasenotes/notes/bump-golang-1.19.4-8a61d93d111cc229.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Agents are now built with Go ``1.19.4``.
```
Comment (vickenty) [inline releasenotes/notes/bump-golang-1.19.4-8a61d93d111cc229.yaml:11]: It feels unusual to mark the version number as an identifier, but feel free to ignore.

```suggestion
    Agents are now built with Go 1.19.4.
```

**Review by vickenty — APPROVED** [review]

**Review by ahmed-mez — APPROVED** [review]

File: releasenotes/notes/bump-golang-1.19.4-8a61d93d111cc229.yaml (hunk 1 above)
Comment (truthbk) [reply releasenotes/notes/bump-golang-1.19.4-8a61d93d111cc229.yaml:11]: I just followed the pattern used in other changelogs in the past; so I'll keep it for consistency if that's OK.

**Review by bkabrda — APPROVED** [review]
:+1: for agent-platform.

**Review by hmahmood — APPROVED** [review]

**Review by hush-hush — APPROVED** [review]

**Review by brycekahle — APPROVED** [review]

**Review by richl-dd — APPROVED** [review]
Approved for WKIT-owned files

**Review by urseberry — APPROVED** [review]

**Review by xornivore — APPROVED** [review]

**Pull request merged**